ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/MainActivity.kt

```diff
@@ -156,10 +210,18 @@ fun WearApp(watchRepository: WatchRepository) {
                         }
                     )
                 ) { navBackStackEntry ->
-                    val watchId =
-                        navBackStackEntry.arguments?.getInt(WATCH_ID_NAV_ARGUMENT)!!
+                    val watchId = navBackStackEntry.arguments?.getInt(WATCH_ID_NAV_ARGUMENT)!!
+                    currentScrollStateKey = Screen.WatchDetail.route + watchId
```

Comment:
This is a side-effect of composition, which is something in general to avoid. I think the reason why this currently works is that this value is always set right before the `PositionIndicator` is reading it, so it queries for the correct value.

Reply:
Oh, good catch, still getting used to the side effect stuff. Let me rewrite it with those APIs.

Reply:
I tried a bunch of different variations that worked, but I tired out the back stack approach you outlined and it worked without the FR you put it. :)

As discussed, we can get sign off from Ian, but I think that's the best solution.